htdocs/install/repair.php

```diff
@@ -1253,6 +1255,54 @@
 	}
 }
 
+// force utf8mb4 on tables  EXPERIMENTAL !
+if ($ok && GETPOST('force_utf8mb4_on_tables', 'alpha')) {
+	print '<tr><td colspan="2"><br>*** Force page code and collation of tables into utf8mb4/utf8mb4_unicode_ci (for mysql/mariadb only)</td></tr>';
+
+	if ($db->type == "mysql" || $db->type == "mysqli") {
+		$force_utf8mb4_on_tables = GETPOST('force_utf8mb4_on_tables', 'alpha');
+
+		$listoftables = $db->DDLListTables($db->database_name);
+
+		// Disable foreign key checking for avoid errors
+		if ($force_utf8mb4_on_tables == 'confirmed') {
+			$sql = 'SET FOREIGN_KEY_CHECKS=0';
+			print '<!-- '.$sql.' -->';
+			$resql = $db->query($sql);
+		}
+
+		foreach ($listoftables as $table) {
+			// do not convert llx_const if mysql encrypt/decrypt is used
+			if ($conf->db->dolibarr_main_db_encryption != 0 && preg_match('/\_const$/', $table)) {
+				continue;
+			}
+
+			print '<tr><td colspan="2">';
+			print $table;
+			$sql = 'ALTER TABLE '.$table.' CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci';
```

Comment:
If you force utf8mb4 on table, you must also force the db engine format to "dynamic" instead of format "compact". If not, the table with a lot of index will generates errors because 4 x size of index must be lower then 750.
An option to mass change from compact to dynamic is also required (and will have to be run before)

Reply:
Before making the conversion into utf8mb4, can you also add a conversion of engine into "dynamic" instead of "contact".
Should be done with "ALTER TABLE table_name ROW_FORMAT=dynamic;" and must be done BEFORE switching to utf8mb4_unicode_ci